src/generic/srchctlg.cpp

```diff
@@ -408,24 +418,20 @@ void wxSearchCtrl::ShowCancelButton( bool show )
         return;
     }
 
-    // This button is not shown initially, so create it on demand if necessary,
-    // i.e. if it's the first time we show it.
-    if ( !m_cancelButton )
+    if ( show )
     {
-        m_cancelButton = new wxSearchButton(this,
-                                            wxEVT_SEARCH_CANCEL,
-                                            m_cancelBitmap);
         RecalcBitmaps();
+        m_cancelButton->Show();
```

Comment:
This entire change (i.e. 4 lines) doesn't seem to be better than the existing version.

Please, please avoid insignificant changes like this in your PRs, they make it much harder to see what has really changed.

Reply:
I am sorry for this one too, it is relative to the above one "Only creating cancel button on demand".